### PR TITLE
[GSoC] Updates stats docs in `doc`

### PR DESCRIPTION
**Pull request by Smit-create**
#### Release Notes

NO ENTRY

### CODE REVIEW

File: doc/src/modules/stats.rst [hunk 1]
```diff
@@ -97,9 +100,22 @@ Joint Types
...
 --------------------
-.. autofunction:: DiscreteMarkovChain
-.. autofunction:: ContinuousMarkovChain
-.. autofunction:: BernoulliProcess
+.. autoclass:: DiscreteMarkovChain
+.. autoclass:: ContinuousMarkovChain
+.. autoclass:: BernoulliProcess
+.. autoclass:: PoissonProcess
+.. autoclass:: WienerProcess
+.. autoclass:: GammaProcess
```
Comment (czgdp1807) [inline doc/src/modules/stats.rst:108]: The reference, https://www.probabilitycourse.com in these three processes should point to their respective pages under the website, for now it is just landing to the home page.

File: doc/src/modules/stats.rst [hunk 2]
```diff
@@ -116,12 +132,25 @@ Interface
...
 .. autofunction:: sample
 .. autofunction:: sample_iter
+.. autofunction:: factorial_moment
+.. autofunction:: kurtosis
+.. autofunction:: skewness
+.. autofunction:: correlation
 .. autofunction:: sympy.stats.rv.sampling_density
 .. autofunction:: sympy.stats.rv.sampling_P
 .. autofunction:: sympy.stats.rv.sampling_E
+.. autoclass:: Moment
```
Comment (czgdp1807) [inline doc/src/modules/stats.rst:147]: New lines missing after, `:` in the following code,
https://github.com/sympy/sympy/blob/4b23967d96ec120dd479fce8e04851a6323f86c3/sympy/stats/symbolic_probability.py#L580-L594

How the docs look like right now?
```python
To evaluate the result of Moment use doit: >>> M.doit() mu**3 - 3*mu**2 + 3*mu*sigma**2 + 3*mu - 3*sigma**2 - 1

Rewrite the Moment expression in terms of Expectation: >>> M.rewrite(Expectation) Expectation((X - 1)**3)

Rewrite the Moment expression in terms of Probability: >>> M.rewrite(Probability) Integral((x - 1)**3*Probability(Eq(X, x)), (x, -oo, oo))

Rewrite the Moment expression in terms of Integral: >>> M.rewrite(Integral) Integral(sqrt(2)*(X - 1)**3*exp(-(X - mu)**2/(2*sigma**2))/(2*sqrt(pi)*sigma), (X, -oo, oo))
```

File: doc/src/modules/stats.rst [hunk 3]
```diff
@@ -116,12 +132,25 @@ Interface
...
+.. autofunction:: moment
+.. autoclass:: CentralMoment
```
Comment (czgdp1807) [inline doc/src/modules/stats.rst:149]: Similar issue here.

File: doc/src/modules/stats.rst (hunk 1 above)
Comment (Smit-create) [reply doc/src/modules/stats.rst:108]: Okay!

File: doc/src/modules/stats.rst (hunk 2 above)
Comment (Smit-create) [reply doc/src/modules/stats.rst:147]: Yes, it will require a new line. Fixed it!

**Comment by czgdp1807** [comment]
LGTM. Just make the tests pass and will merge it.

**Pull request merged**